server/functions/index.js

```diff
@@ -288,3 +284,47 @@ exports.sendNotificationOnNewChat = functions.firestore
         }
     });
 
+// delete all favors when a user is deleted
+exports.deleteFavorsOnUserDeletion = functions.firestore
+    .document('users/{userId}')
+    .onDelete((snap) => {
+
+        const deletedUserId = snap.data().id;
+
+        // go through all the users
+        return db.collection('/favors').where('userIds', 'array-contains', deletedUserId).get()
+            .then((snapshot) => {
+                snapshot.forEach((doc) => {
+                    const favor = doc.data();
+
+                    if (favor.userIds.length === 1) {
+                        doc.ref.delete().then(() => {
+                            console.log("Favor successfully deleted!");
+                        }).catch(function (error) {
+                            console.error("Error deleting favor: ", error);
+                        });
+                    }
+                });
+            })
+    });
+
+
+// delete all chats messages when a favor is deleted
+exports.deleteChatOnFavorDeletion = functions.firestore
+    .document('favors/{favorId}')
+    .onDelete((snap) => {
+
+        const deletedFavorId = snap.data().id;
+
+        // go through all the chat messages
+        return db.collection('/chats').where('favorId', '==', deletedFavorId).get()
+            .then((snapshot) => {
+                snapshot.forEach((doc) => {
+                    doc.ref.delete().then(() => {
```

Comment:
not sure if should return this promise for the last iteration to ensure it finishes

Reply:
Sorry I don't understand the problem, why does returning this promise for the last iteration is a problem?